llvm/lib/Target/AMDGPU/SOPInstructions.td

```diff
@@ -2012,6 +2012,14 @@ let AddedComplexity = 20 in {
   >;
 }
 
+let SubtargetPredicate = isGFX9Plus in
+foreach I = 1-4 in {
+def : GCNPat <
+  (i32 (UniformBinFrag<add> (shl_oneuse i32:$src0, (i32 I)), i32:$src1)),
+  (!cast<SOP2_Pseudo>("S_LSHL"#I#"_ADD_U32") $src0, $src1)
```

Comment:
There is already the same pattern on the instruction definition, so why isn't that one working? Shouldn't need to repeat this here

Reply:
So can just remove that then